larray/inout/csv.py

```diff
@@ -223,26 +253,58 @@ def _open_for_write(self):
             except OSError:
                 if not os.path.isdir(self.directory):
                     raise ValueError("Path {} must represent a directory".format(self.directory))
+        self._load_axes_and_groups()
```

Comment:
This shouldn't be necessary (only initializing self.axes and self.groups to OrderedDict should be necessary) and is potentially dangerous as this will (AFAICT) keep old "unused" axes and groups around, which is not what you want in 99% of cases.